components/raftstore/src/store/async_io/write_router.rs

```diff
@@ -163,13 +167,14 @@ where
         if self.last_unpersisted.is_some() {
             return false;
         }
-        if ctx.config().store_io_pool_size <= 1 {
-            self.writer_id = 0;
-            return true;
-        }
+        // Local senders may not be updated when `store_io_pool_size()` has been
+        // increased by the `ctx.config().update()`, keep the real size until it's
+        // updated by `poller.begin()`.
+        let async_io_pool_size =
+            std::cmp::min(ctx.write_senders().size(), ctx.config().store_io_pool_size);
```

Comment:
add an assertion `config.io_size <= senders().size()`

Reply:
It's incorrect when increasing `store-io-pool-size`.

In this case, the `config.store-io-pool-size` will be updated firstly, followed by the triggering of increasing async-ios by the controller worker.